src/store/actions.js

```diff
@@ -148,6 +148,10 @@ export default {
 			return account
 		})
 	},
+	setAccountSetting({ commit, getters }, { accountId, key, value }) {
+		commit('setAccountSetting', { accountId, key, value })
+		savePreference('account-settings', JSON.stringify(getters.getAllAccountSettings))
```

Comment:
We could use `Object.values()` indeed to strip the IDs

Reply:
Thanks for the feedback! I just noticed that doing a rebase right away is not the best for incremental reviews... 🙄 

Anyway, I decided to save it as an array of objects now, instead of an object with stringified ids. So the stored array now looks like this:

```json
[{"accountId":75,"collapsed":true},{"accountId":76,"collapsed":false}]
```

When merging the settings with the account (in main.js), I am simply grabbing the proper ones using `find`:

```javascript
const settings = accountSettings.find(settings => settings.accountId === account.id)
```

Most notably, [`setAccountSetting`](https://github.com/nextcloud/mail/pull/3952/commits/a150c9867256abb1a17ed4021d29c8f3a5285dbd#diff-7c0816590b8c43968bdb997a4d7ab7a803f17f160a132376e0b3aa1c9859d2f5R113-R122) has changed quite a bit in the process:

```javascript
setAccountSetting(state, { accountId, key, value }) {
	const accountSettings = state.allAccountSettings.find(settings => settings.accountId === accountId)
	if (accountSettings) {
		accountSettings[key] = value
	} else {
		const newAccountSettings = { accountId }
		newAccountSettings[key] = value
		state.allAccountSettings.push(newAccountSettings)
	}
},
```

... which takes care of updating existing settings or pushing new settings if there haven't been any for an account yet.

Lastly, I changed `state.accountSettings` to `state.allAccountSettings` for consistency.

Reply:
The settings are saved to the database and persist one reload. However, `state.allAccountSettings` is not set on load in `main.js` (only the account state via `addAccount()`).

You could do something like:

```diff
diff --git a/src/main.js b/src/main.js
index 409b7f6f..018906c2 100644
--- a/src/main.js
+++ b/src/main.js
@@ -76,6 +76,16 @@ const accountSettings = JSON.parse(Base64.decode(getPreferenceFromPage('account-
 const accounts = JSON.parse(Base64.decode(getPreferenceFromPage('serialized-accounts')))
 accounts.map(fixAccountId).forEach((account) => {
 	const settings = accountSettings.find(settings => settings.accountId === account.id)
+	if (settings) {
+		delete settings.accountId
+		Object.entries(settings).forEach(([key, value]) => {
+			store.commit('setAccountSetting', {
+				accountId: account.id,
+				key,
+				value,
+			})
+		})
+	}
 	store.commit('addAccount', { ...account, ...settings })
 })
 

```

Reply:
Thanks Richard, I'll take that as-is. :-)

I made the change and pulled in the latest changes from master.